### PR TITLE
ISSUE-147 isChannelClosed should be public

### PR DESCRIPTION
**Pull request by chibenwa**
## Description

I would like to assert the state of a session before submitting requests to it.

## Motivation and Context

IMAP provisionning script

## How Has This Been Tested?

All tests passes

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Major release (change is NOT backward compatible with prior release)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by chibenwa** [comment]
Hello @fansu 

What's the ETA of this?

File: core/src/main/java/com/yahoo/imapnio/async/internal/ImapAsyncSessionImpl.java [hunk 1]
```diff
@@ -356,7 +356,7 @@ public <T> ImapFuture<ImapAsyncResponse> startCompression() throws ImapAsyncClie
     /**
      * @return true if channel is closed; false otherwise
      */
-    boolean isChannelClosed() {
+    public boolean isChannelClosed() {
```
Comment (fansu) [inline core/src/main/java/com/yahoo/imapnio/async/internal/ImapAsyncSessionImpl.java:359]: could you add unit test coverage for isChannelClosed in ImapAsyncSessionImplTest.java

**Review by fansu — CHANGES_REQUESTED** [review]
comments

**Comment by chibenwa** [comment]
The last commit is an attempt to keep the test file below 2000 line of code and not trigger a checkstyle violation.

**Review by fansu — APPROVED** [review]

**Pull request merged**